pkg/rtc/types/interfaces.go

```diff
@@ -329,6 +326,11 @@ type LocalParticipant interface {
 	SetSignalSourceValid(valid bool)
 	HandleSignalSourceClose()
 
+	// updates
+	SetName(name string)
+	SetMetadata(metadata string)
+	SetAttributes(attributes map[string]string) error
```

Comment:
These are only required on the local participant. remote participants are not updatable.